tableauserverclient/models/custom_view_item.py

```diff
@@ -0,0 +1,156 @@
+        return self._shared
+
+    @shared.setter
+    def shared(self, value: bool):
+        self._shared = value
+
+    @property
+    def updated_at(self) -> Optional["datetime"]:
+        return self._updated_at
+
+    @property
+    def owner(self) -> Optional[UserItem]:
+        return self._owner
+
+    @owner.setter
+    def owner(self, value: UserItem):
+        self._owner = value
+
+    @property
+    def workbook(self) -> Optional[WorkbookItem]:
+        return self._workbook
+
+    @property
+    def view(self) -> Optional[ViewItem]:
+        return self._view
+
+    @classmethod
+    def from_response(cls, resp, ns, workbook_id="") -> Optional["CustomViewItem"]:
+        item = cls.list_from_response(resp, ns, workbook_id)
+        if not item or len(item) == 0:
+            return None
+        else:
+            return item[0]
+
+    @classmethod
+    def list_from_response(cls, resp, ns, workbook_id="") -> List["CustomViewItem"]:
+        return cls.from_xml_element(fromstring(resp), ns, workbook_id)
+
+    """
+    <customView
+    id="37d015c6-bc28-4c88-989c-72c0a171f7aa"
+    name="New name 2"
+    createdAt="2016-02-03T23:35:09Z"
+    updatedAt="2022-09-28T23:56:01Z"
+    shared="false">
+      <view id="8e33ff19-a7a4-4aa5-9dd8-a171e2b9c29f" name="circle"/>
+      <workbook id="2fbe87c9-a7d8-45bf-b2b3-877a26ec9af5" name="marks and viz types 2"/>
+      <owner id="cdfe8548-84c8-418e-9b33-2c0728b2398a" name="workgroupuser"/>
+    </customView>
+    """
+
+    @classmethod
+    def from_xml_element(cls, parsed_response, ns, workbook_id="") -> List["CustomViewItem"]:
+        all_view_items = list()
+        all_view_xml = parsed_response.findall(".//t:customView", namespaces=ns)
+        for custom_view_xml in all_view_xml:
+            cv_item = cls()
+            view_elem: ElementTree = custom_view_xml.find(".//t:view", namespaces=ns)
+            workbook_elem: str = custom_view_xml.find(".//t:workbook", namespaces=ns)
+            owner_elem: str = custom_view_xml.find(".//t:owner", namespaces=ns)
+            cv_item._created_at = parse_datetime(custom_view_xml.get("createdAt", None))
+            cv_item._updated_at = parse_datetime(custom_view_xml.get("updatedAt", None))
+            cv_item._content_url = custom_view_xml.get("contentUrl", None)
+            cv_item._id = custom_view_xml.get("id", None)
+            cv_item._name = custom_view_xml.get("name", None)
+
+            if owner_elem is not None:
+                parsed_owners = UserItem.from_response_as_owner(tostring(custom_view_xml), ns)
+                if parsed_owners and len(parsed_owners) > 0:
+                    cv_item._owner = parsed_owners[0]
+
+            if view_elem is not None:
+                parsed_views = ViewItem.from_response(tostring(custom_view_xml), ns)
+                if parsed_views and len(parsed_views) > 0:
+                    cv_item._view = parsed_views[0]
+
+            if workbook_id:
```

Comment:
For my knowledge, what is the reasoning behind favoring this `workbook_id` override instead of always parsing the XML response?

Reply:
This is matching the behavior in view_item.py, where I believe the reasoning was       
a) our Fields implementation means a View response doesn't have to contain a workbook element, but the user quite likely has the workbook id available     
 and b) if both the parameter and the xml element are available, we want a consistent priority order and chose this order as least likely to surprise a user: if they pass in a value, they will know where it came from when it shows up.